### PR TITLE
Fix field of a type that has a default value

### PR DESCRIPTION
**Pull request by koxudaxi**
## Change Summary

The PR fixes the field of a type that has a default value.

## Related issue number
https://github.com/samuelcolvin/pydantic/issues/878

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

File: tests/test_edge_cases.py [hunk 1]
```diff
@@ -957,15 +957,19 @@ def __init__(self, **data) -> None:
...
-
-    assert Model.__fields__.keys() == {'b'}
+        a: int = int
+        b: Type[int]
+        c: Type[int] = int
+        d: FooBar = FooBar
+        e: Type[FooBar]
+        f: Type[FooBar] = FooBar
+
+    assert Model.__fields__.keys() == {'b', 'c', 'e', 'f'}
```
Comment (dmontagu) [inline tests/test_edge_cases.py]: Any chance you could add a test like what you had in the issue, where you were explicitly confirmed the value of the set property was wrong? Feels like it's worth explicitly checking the behavior is being handled properly

Comment (koxudaxi) [reply tests/test_edge_cases.py]: Thank you for your review. I add a test case.

Comment (dmontagu) [reply tests/test_edge_cases.py]: This looks great! Thanks!

File: tests/test_edge_cases.py [hunk 2]
```diff
@@ -957,15 +957,48 @@ def __init__(self, **data) -> None:
...
+
+    class Parent:
```
Comment (samuelcolvin) [inline tests/test_edge_cases.py]: From here on I think this should be a separate test (you'll need to define `Model` with the fields required).

**Comment by koxudaxi** [comment]
I have separated the test.

**Comment by samuelcolvin** [comment]
great, thank you.

**Comment by koxudaxi** [comment]
Thank you!!

**Pull request merged**